nesi_jupyter_helpers/add_kernel.py

```diff
@@ -130,15 +161,34 @@ def add_kernel(
                 f"ERROR: --venv ({venv}) does not appear to be a virtual "
                 "environment (cannot find bin/activate)"
             )
-        venv_txt = VENV_TEMPLATE.format(venv_activate_script=venv_activate_script)
+        exec_txt = VENV_TEMPLATE.format(venv_activate_script=venv_activate_script)
         if not any(m.startswith("Python") for m in module):
             print(
                 "WARNING: Make sure to specify the appropriate Python module "
                 "for your virtual environment."
             )
 
+    # ...or a Singularity container...
+    elif container is not None:
+        container = container.resolve()
+        if shared:
+            print("Make sure your container is accessible to members of {account}")
+        if not container.is_file():
+            sys.exit(
+                f"ERROR: --container ({container}) should point to a Singularity "
+                "container image file"
+            )
+        runtime_dir = Path.home().resolve() / ".local/share/jupyter/runtime"
```

Comment:
If you want to bind just the runtime directory and not the whole resolved home directory path (which might be safer, perhaps we should do both?) then I think it would be best to get the runtime path from jupyter_core instead of hardcoding it, e.g.

```python
import jupyter_core.paths
runtime_dir = jupyter_core.paths.jupyter_runtime_dir()
```

Reply:
Brilliant! I'll update the code.